providers/zoho.go

```diff
@@ -0,0 +1,82 @@
+package providers
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/pusher/oauth2_proxy/pkg/apis/sessions"
+	"github.com/pusher/oauth2_proxy/pkg/requests"
+)
+
+// ZohoProvider represents a Zoho based Identity Provider
+type ZohoProvider struct {
+	*ProviderData
+}
+
+// NewZohoProvider initiates a new ZohoProvider
+func NewZohoProvider(p *ProviderData) *ZohoProvider {
+	p.ProviderName = "Zoho"
+	if p.LoginURL.String() == "" {
+		p.LoginURL = &url.URL{Scheme: "https",
+			Host:     "accounts.zoho.com",
+			Path:     "/oauth/v2/auth",
+			RawQuery: "response_type=code&access_type=offline",
```

Comment:
I've not seen `RawQuery` set in one of the providers before, I noticed that this duplicates the call to set `response_type` [here](https://github.com/pusher/oauth2_proxy/blob/9670f54dd00fd75ffd0fb765f8fd60aa64c1fabd/providers/provider_default.go#L92). But the `access_type` is new. Can you verify for me that all the parameters are set correctly in by generating a login URL?  Is there a test for the login URL?